crates/orchestrator/src/lib.rs

```diff
@@ -179,6 +207,57 @@ where
         Ok(self.config.clone())
     }
 
+    #[allow(clippy::cast_possible_truncation)]
+    fn register_public_key(
+        &mut self,
+        node_index: u64,
+        pubkey: &mut Vec<u8>,
+    ) -> Result<(), ServerError> {
+        if self.pub_posted.contains(&node_index) {
+            return Err(ServerError {
+                status: tide_disco::StatusCode::BadRequest,
+                message: "Node has already posted public key".to_string(),
+            });
+        }
+        self.pub_posted.insert(node_index);
+
+        // Sishan NOTE: let me know if there's a better way to remove the first extra 8 bytes
+        pubkey.drain(..8);
```

Comment:
Makes sense to use `drain`! Can you add a comment about why we want to delete these bytes?

Reply:
Sure, added! Actually I'm not 100% sure about why, my guess is orchestrator did their own serialization, therefore it adds 8 more bytes for the type configuration.